astropy/cosmology/_src/traits/baryons.py

```diff
@@ -15,7 +15,30 @@
 
 
 class BaryonComponent:
-    """The cosmology has attributes and methods for the baryon density."""
+    """The cosmology has attributes and methods for the baryon density.
+
+    Examples
+    --------
+    For an example of a real cosmology that implements this trait, see
+    :class:`~astropy.cosmology.LambdaCDM`. Here we will define an illustrative example
+    class that meets the minimum API requirements, but is not cosmologically meaningful:
+
+    >>> from astropy.cosmology.traits import BaryonComponent
+    >>> import dataclasses
+
+    >>> @dataclasses.dataclass(frozen=True)
+    ... class ExampleHasBaryons(BaryonComponent):
+    ...     Ob0: float
```

Comment:
It seems annoying that you would have to define `Ob0` when the trait already does so. Doesn't the order of traits imply an order here?

Reply:
It's not strictly necessary, just often good practice when making a dataclass to declare all fields for exactly the reason you mention: the order of traits implies the order of the fields. Being very explicit can be useful in that regard.